### PR TITLE
Restrict Levels to categorical columns

### LINKED ISSUE #95: Consider restricting Levels to columns with categorical scitype
Currently the Levels transform contributed by @Frank-III and refactored by @eliascarv can operate directly on any type of column, including columns with the wrong scitype. I think we could consider a more consistent approach where we ask users to be explicit about the scitypes using the Coerce transform first. Even though this is a breaking change, it will provide more safety in real-world applications of pipelines.

### PR DESCRIPTION
**Pull request by eliascarv**
closes #95

### CODE REVIEW

File: src/transforms/levels.jl [hunk 1]
```diff
@@ -29,28 +29,24 @@ Levels(; kwargs...) = throw(ArgumentError("Cannot create a Levels object without
...
+      assert_categorical(x)
+      
       o = nm ∈ ordered
-      l = levels[findfirst(==(nm), snames)]
-      return _categorical(x, l, o)
+      l = tlevels[findfirst(==(nm), snames)]
+      y = categorical(x, levels=l, ordered=o)
+      xl, xo = levels(x), isordered(x)
+      revfunc = y -> categorical(y, levels=xl, ordered=xo)
+      return y, revfunc
```
Comment (juliohm) [inline src/transforms/levels.jl:49]:
```suggestion
      y, revfunc
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Small adjustment

File: src/transforms/levels.jl (hunk 1 above)
Comment (eliascarv) [reply src/transforms/levels.jl:49]: This `return` is necessary.

Comment (juliohm) [reply src/transforms/levels.jl:49]: Oh so maybe we are missing an else branch in the if statement? I reviewed the code quickly and missed that this was an actual case of early termination.

Comment (eliascarv) [reply src/transforms/levels.jl]: Take a look at the new version of the code, @juliohm.

**Review by juliohm — APPROVED** [review]

**Pull request merged**